sim/core/spell.go

```diff
@@ -223,13 +207,13 @@ func ApplyEffectFuncDamageMultiple(baseEffects []SpellEffect) ApplySpellEffects
 	}
```

Comment:
These changes were checked in rather accidentally, but remove another ~10% (amount-wise) allocations. They might clash with the two "OnInit()" cases (Sunder Armor and Bloodthirst), but it probably makes sense to remove those two special cases alltogether (since threat calculation isn't part of the tests, there's no failures here).